crates/iota-core/src/authority.rs

```diff
@@ -861,7 +861,7 @@ impl AuthorityState {
         epoch_store: &Arc<AuthorityPerEpochStore>,
     ) -> IotaResult<VerifiedSignedTransaction> {
         // Ensure that validator cannot reconfigure while we are signing the tx
-        let _execution_lock = self.execution_lock_for_signing().await;
+        let _execution_lock = self.execution_lock_for_signing();
```

Comment:
Looks like the current fn `handle_transaction_impl` is also non-async. You can run `cargo clippy -- -D clippy::unused_async` to check for it.

Reply:
We might fix that in another dedicated PR.